### ISSUE TITLE
SNS publish: invalid parameter TargetArn

### ISSUE THREAD

**Issue by jaimeiniesta** [CargoSense/ex_aws#402]
Hello, I've been able to create a topic in SNS but I'm not able to publish to it, what am I doing wrong?

```elixir
iex(11)> ExAws.SNS.publish("hello friends", [TargetArn: "arn:aws:sns:eu-west-1:686871504305:92B39B61-CCAE-4C4E-869A-D05EF4001EC8_topic"]) |> ExAws.request
{:error,
 {:http_error, 400,
  %{body: "<ErrorResponse xmlns=\"http://sns.amazonaws.com/doc/2010-03-31/\">\n  <Error>\n    <Type>Sender</Type>\n    <Code>InvalidParameter</Code>\n    <Message>Invalid parameter: TargetArn</Message>\n  </Error>\n  <RequestId>e32df0c3-7e41-50cc-bb21-a1492e54b12b</RequestId>\n</ErrorResponse>\n",
    headers: [{"x-amzn-RequestId", "e32df0c3-7e41-50cc-bb21-a1492e54b12b"},
     {"Content-Type", "text/xml"}, {"Content-Length", "275"},
     {"Date", "Mon, 08 May 2017 10:52:05 GMT"}, {"Connection", "close"}],
    status_code: 400}}}
```

This will fail both when passing the `TargetArn` or `TopicArn` parameter.

**Comment by benwilson512** [comment]
Hey there, the docs for that function show that the options are `target_arn` and `topic_arn`. Do those work for you? https://hexdocs.pm/ex_aws/ExAws.SNS.html#t:publish_opts/0

**Comment by jaimeiniesta** [comment]
Thanks, this is what I tried in the first place, using `target_arn` but didn't work:

```elixir
iex(5)> ExAws.SNS.publish("hello friendsters", [target_arn: "arn:aws:sns:eu-west-1:686871504305:92B39B61-CCAE-4C4E-869A-D05EF4001EC8_topic"]) |> ExAws.request
{:error,
 {:http_error, 400,
  %{body: "<ErrorResponse xmlns=\"http://sns.amazonaws.com/doc/2010-03-31/\">\n  <Error>\n    <Type>Sender</Type>\n    <Code>InvalidParameter</Code>\n    <Message>Invalid parameter: TargetArn</Message>\n  </Error>\n  <RequestId>6ba94cd8-38b4-5970-91bc-1d526243aa03</RequestId>\n</ErrorResponse>\n",
    headers: [{"x-amzn-RequestId", "6ba94cd8-38b4-5970-91bc-1d526243aa03"},
     {"Content-Type", "text/xml"}, {"Content-Length", "275"},
     {"Date", "Mon, 08 May 2017 11:38:30 GMT"}, {"Connection", "close"}],
    status_code: 400}}}
```

Then looking at other issues I saw examples using `TargetArn`. In any case they're properly camel cased later, so both do the same in the end.

But I just found something weird. This topic was created with `ExAws`, and from the web console in AWS I see it has this arn in an European zone (which is fine for me, I'm in Spain)

`arn:aws:sns:eu-west-1:686871504305:92B39B61-CCAE-4C4E-869A-D05EF4001EC8_topic`

But then if I list the topics it looks like it's on an US zone:

```elixir
iex(6)> ExAws.SNS.list_topics() |> ExAws.request!
%{body: "<ListTopicsResponse xmlns=\"http://sns.amazonaws.com/doc/2010-03-31/\">\n  <ListTopicsResult>\n    <Topics>\n      <member>\n        <TopicArn>arn:aws:sns:us-east-1:686871504305:deleteme</TopicArn>\n      </member>\n    </Topics>\n  </ListTopicsResult>\n  <ResponseMetadata>\n    <RequestId>0f023dc5-092e-585e-9866-6c4f52447c50</RequestId>\n  </ResponseMetadata>\n</ListTopicsResponse>\n",
  headers: [{"x-amzn-RequestId", "0f023dc5-092e-585e-9866-6c4f52447c50"},
   {"Content-Type", "text/xml"}, {"Content-Length", "373"},
   {"Date", "Mon, 08 May 2017 11:40:26 GMT"}], status_code: 200}
```

And if I use this other arn in the US, then it works fine:

```elixir
iex(3)> ExAws.SNS.publish("hello friends", [TargetArn: "arn:aws:sns:us-east-1:686871504305:deleteme"]) |> ExAws.request
{:ok,
 %{body: "<PublishResponse xmlns=\"http://sns.amazonaws.com/doc/2010-03-31/\">\n  <PublishResult>\n    <MessageId>3f7b7de3-3706-501e-ab01-2c696b815ccd</MessageId>\n  </PublishResult>\n  <ResponseMetadata>\n    <RequestId>0dcd44c7-a7ca-52f5-a2b9-154dfb4d4e26</RequestId>\n  </ResponseMetadata>\n</PublishResponse>\n",
   headers: [{"x-amzn-RequestId", "0dcd44c7-a7ca-52f5-a2b9-154dfb4d4e26"},
    {"Content-Type", "text/xml"}, {"Content-Length", "294"},
    {"Date", "Mon, 08 May 2017 11:45:11 GMT"}], status_code: 200}}
iex(4)>
```

So to sum it up it looks like `publish` works fine but there's something weird with the `arn`s and zones. Let me clear up my mind and come with a more clear issue later.

**Comment by benwilson512** [comment]
What region are you running your requests in, US? If you're trying to publish to a an EU target you quite probably need to be making requests targeting the EU region.

**Comment by jaimeiniesta** [comment]
Thank you, I've checked again and everything works fine - I was confused about the region, that's all :)
